### DISCUSSION TITLE
It appears that there was a problem signing your application prior to installation on the device.

### DISCUSSION THREAD

**Discussion by timfong888** [codemagic-ci-cd/codemagic-docs#2250] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

I pushed a Github repo and tried to build for IOS only.

It appears that there was a problem signing your application prior to installation on the device.

Verify that the Bundle Identifier in your project is your signing id in Xcode
  open ios/Runner.xcworkspace

Also try selecting 'Product > Build' to fix the problem.
Encountered error while archiving for device.

### Expected results

I expect it to build and be able to run is on TestFlight

### Actual results

It appears that there was a problem signing your application prior to installation on the device.

Verify that the Bundle Identifier in your project is your signing id in Xcode
  open ios/Runner.xcworkspace

Also try selecting 'Product > Build' to fix the problem.
Encountered error while archiving for device.

### Build id (optional)

65715660fa321dd2266b390f

**Comment by dtrdic** [comment]
hey @timfong888, i can see this kind of error in the logs
```
Error (Xcode): Provisioning profile doesn't support the Push Notifications capability.
Error (Xcode): Provisioning profile doesn't include the aps-environment entitlement. Profile qualification is using entitlement definitions that may be out of date
```

Xcode is trying to archive `sophia.pkg` bundleID. Have you added capabilities for that identifier on apple developer side?
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/da50219d-b7f8-4c5f-8a8d-ddbbd81ca7bc)


Have you added capabilities inside your xcode project as well?
![Screenshot 2023-12-07 at 13 16 10](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/3fb855ac-04fa-48cf-a098-ee57e83d8a61)

**Reply by timfong888** [reply]
Hi, I added the capability in my App and am using Flutterflow for pushing to Testflight. Given the capability is there, I'm not sure why there was an issue.

**Comment by icarusdust** [comment]
While I am closing this thread @timfong888 due to inactivity, I need to point out that your provisioning profile type is `development` and you enabled App Store Connect publishing.  For ASC publishing, you will need to choose `App Store` profile type by going to the Workflow settings and expanding the iOS code signing dropdown.

**Reply by timfong888** [reply]
Oh, I see, I didn't know I needed that profile type.  I just saw an email where I can push to Testflight...that's what I am doing right now....it was in video, if there's something in written docs, that could help my understanding of the video.
